core/src/main/java/org/apache/calcite/sql/validate/DelegatingScope.java

```diff
@@ -297,8 +298,17 @@ protected void addColumnNames(
         namespace = map.get(tableName).namespace;
         break;
       default:
-        throw validator.newValidationError(identifier,
-            RESOURCE.columnAmbiguous(columnName));
+        // Allow if all namespaces are the same and the tables are not null
```

Comment:
I think that when this PR is merged, the column ambiguous problem will be solved, so no changes are needed here.
https://github.com/apache/calcite/pull/3797

Reply:
@normanj-bitquill here, is this modification necessary?

Reply:
@YiwenWu I'm not sure. If we have a query such as:

```
select x, * from foo f join bar b using (x)
```

Does it matter if we choose `foo.x` or `bar.x` in this context? Due to the join, they should have the same value.

Reply:
Without this modification, verified that the two examples of unit tests can also be executed successfully.

Reply:
@YiwenWu  Rebased this change off of the current main branch. As a result, this change is not needed and has been removed.